tests/overmap_test.cpp

```diff
@@ -19,3 +21,30 @@ TEST_CASE( "set_and_get_overmap_scents" ) {
     REQUIRE( test_overmap.scent_at( { 75, 85, 0} ).creation_turn == 50 );
     REQUIRE( test_overmap.scent_at( { 75, 85, 0} ).initial_strength == 90 );
 }
+
+TEST_CASE( "default_overmap_generation_always_succeeds" ) {
+    // List of specials to generate, overmap should consume them all.
+    std::vector<const overmap_special *> default_specials = overmap( 0, 0 ).get_enabled_specials();
+    int overmaps_to_construct = 10;
+    for( point candidate_addr : closest_points_first( 10, { 0, 0 } ) ) {
+         // Skip populated overmaps.
+      if( overmap_buffer.has( candidate_addr.x, candidate_addr.y ) ) {
+            continue;
+        }
+        std::vector<overmap_special_placement> test_specials;
+        for( auto special : default_specials ) {
+            test_specials.push_back( { 0, special } );
+        }
+        overmap_buffer.create_custom_overmap( candidate_addr.x, candidate_addr.y, test_specials );
```

Comment:
I'm not sure if I understand what this function does.
Does it overwrite the specials or not?
Reading the function, it looks to me like it drops the whole overmap and generates a new one, with the specials from argument. If so, it doesn't really test that the specials are passed around properly and don't "overflow".